test/plugin/base_test.rb

```diff
@@ -1020,6 +1020,134 @@ def test_http_request_skip_setting_latency
     end
   end
 
+  def test_source_location_from_record
```

Comment:
Similar here. Can we move the common logic to verify these fields to shared methods? Then the test itself simply specify the dynamic part (input and output).

e.g.
```
def verify_log_entry_subfield(.., .., ..)
...
end

def test_log_entry_operation_field
   ...
   [<SOME_DATA_TO_TEST_FROM_RECORD>, <SOME_DATA_TO_TEST_PARTIAL>, <SOME_DATA_TO_TEST_NON_HASH>].each do |...
     verify_log_entry_subfield(...
   end
    ...
end

def test_log_entry_source_location_field
   ...
   [<SOME_DATA_TO_TEST_FROM_RECORD>, <SOME_DATA_TO_TEST_PARTIAL>, <SOME_DATA_TO_TEST_NON_HASH>].each do |...
     verify_log_entry_subfield(...
   end
    ...
end

def test_log_entry_trace_field
   ...
   [<SOME_DATA_TO_TEST_FROM_RECORD>, <SOME_DATA_TO_TEST_PARTIAL>, <SOME_DATA_TO_TEST_NON_HASH>].each do |...
     verify_log_entry_subfield(...
   end
    ...
end